group-coordinator/src/main/java/org/apache/kafka/coordinator/group/Utils.java

```diff
@@ -349,12 +349,19 @@ static void throwIfRegularExpressionIsInvalid(
      * @return The hash of the group.
      */
     static long computeGroupHash(Map<String, Long> topicHashes) {
-        if (topicHashes.isEmpty()) {
+        // Sort entries by topic name
+        List<Map.Entry<String, Long>> sortedEntries = new ArrayList<>();
+        for (Map.Entry<String, Long> entry : topicHashes.entrySet()) {
+            // Filter out entries with a hash value of 0, which indicates no topic
+            if (entry.getValue() != 0) {
```

Comment:
Ignore topic hash 0 because it indicates the topic is not existent. We also do this in previous implementation.

https://github.com/apache/kafka/blob/cff10e654149ed2d4924235a15156d4bc0bf7b4a/group-coordinator/src/main/java/org/apache/kafka/coordinator/group/modern/ModernGroup.java#L387-L396

Reply:
Sure, I understand that we may have nonexistent topics here. I was more trying to understand whether having those zeros in the final hash was an issue. I suppose that you're saying that it is an issue.

Reply:
btw, it seems that we could move the zero check to L369 to simplify a bit the code.

Reply:
Removed related change here, because we can ignore null `TopicImage` in `ModernGroup#computeMetadataHash`. Thanks.